### PR TITLE
fix: prevent cloning menu

### PR DESCRIPTION
**Pull request by picodoth**

### CODE REVIEW

**Comment by JonDum** [comment]
Tried this pull out in my project and confirmed it fixed the `react-router` context issue 👍

**Comment by yesmeck** [comment]
>     ReactDOM.render(this.props.overflowedIndicator, container, () => {


`this.props.overflowedIndicator` 是 ReactNode 话还是会被 mount 两次？

**Comment by yesmeck** [comment]
screen shot 2018-07-17 at 10 50 08

**Comment by picodoth** [comment]
@yesmeck @chenshuai2144  再看下：

- [ ] ... 也只是 mount 一次
- [ ] ... overflow menu 需要应用 theme 样式
- [ ] ... overflow menu 覆盖问题

File: src/DOMWrap.jsx [hunk 1]
```diff
@@ -40,100 +28,73 @@ class DOMWrap extends React.Component {
...
-        if (!ul) {
-          return;
-        }
 
-        const scrollWidth = getScrollWidth(ul);
+    this.childrenSizes = [];
+    const { children } = this.props;
 
-        this.props.children.forEach((c, i) => this.childrenSizes[i] = getWidth(ul.children[i]));
+    children.forEach((c, i) => this.childrenSizes[i] = getWidth(ul.children[i]));
```
Comment (yesmeck) [inline src/DOMWrap.jsx]: `map`

**Comment by yesmeck** [comment]
放到 antd 里再测下

**Comment by picodoth** [comment]
7.2.5

File: src/DOMWrap.jsx [hunk 2]
```diff
@@ -40,100 +28,73 @@ class DOMWrap extends React.Component {
...
     return (
       <SubMenu
         title={overflowedIndicator}
-        className={`${this.props.prefixCls}-overflowed-submenu`}
+        className={`${prefixCls}-overflowed-submenu`}
+        popupClassName={popupClassName}
         {...rest}
         eventKey="overflowed-indicator"
         disabled={false}
+        style={style}
```
Comment (afc163) [inline src/DOMWrap.jsx]: props.style 丢了

File: src/SubPopupMenu.js [hunk 3]
```diff
@@ -343,10 +342,13 @@ export class SubPopupMenu extends React.Component {
         prefixCls={prefixCls}
         mode={mode}
         tag="ul"
+        level={level}
+        theme={theme}
         hiddenClassName={`${prefixCls}-hidden`}
         visible={visible}
         overflowedIndicator={overflowedIndicator}
         {...domProps}
+        style={{ position: 'relative', zIndex: level }}
```
Comment (afc163) [inline src/SubPopupMenu.js]: props.style 丢了

**Pull request merged**